core/bind/core_bind.h

```diff
@@ -660,4 +660,50 @@ class _Engine : public Object {
 	_Engine();
 };
 
+class _JSON;
+
+class JSONParseResult : public Reference {
+	GDCLASS(JSONParseResult, Reference)
+
+	friend class _JSON;
+
+	Error error;
+	String error_string;
+	int error_line;
+
+	Variant result;
+
+protected:
+	static void _bind_methods();
+
+public:
+	void set_error(Error p_error);
+	Error get_error() const;
+
+	void set_error_string(const String &p_error_string);
+	String get_error_string() const;
+
+	void set_error_line(int p_error_line);
+	int get_error_line() const;
+
+	void set_result(const Variant &p_result);
+	Variant get_result() const;
+};
+
+class _JSON : public Object {
+	GDCLASS(_JSON, Object)
+
+protected:
+	static void _bind_methods();
+	static _JSON *singleton;
+
+public:
+	static _JSON *get_singleton() { return singleton; }
+
+	String print(const Variant &p_value);
```

Comment:
I don't think `print` is a good name for the encoding function. Python has it as `dumps`/`loads`, Javascript is with `stringify`/`parse`, in Go it is `Marshall`/`Unmarshall`, PHP goes with `encode`/`decode`.

Reply:
I just went with the names of the `JSON` class, I'm not married to those names 😄

Reply:
Well, go with encode/decode then? 😆

Reply:
My current code is Approved By Reduz ™️ , if the naming confuses people they can tell us, I think it's not that bad.